### DISCUSSION TITLE
Multiple subscriptions

### DISCUSSION THREAD

**Discussion by cloudnowjuha** [shibayan/appservice-acmebot#356] [Q&A] [answered]
Is it possible to implement appservice-acmebot so, that it could serve multiple different subscriptions and app services in all of those subscriptions from only one deployment? If this is now not possible, might be a good development idea to have Subscription as one of the variables that could be set during the add-certificate process.

If I have DNS zones implemented in only one subscription, does the acmebot function app handle Azure DNS related tasks, or do I need to have Azure DNS in same subscription? Now I am getting DNS01Precondition failed error message, even that I have DNS Zone Contributor role specified for function app as needed.

**Comment by shibayan** [comment, marked as answer]
Multiple subscriptions are not supported. You must have an App Service or Azure DNS for the subscription specified by `Acmebot:SubscriptionId`.

This is because the Azure REST API always requires a single subscription.

**Reply by cloudnowjuha** [reply]
Thanks for your answer, I will need to somehow work around this then.

Could this product possibly be developed so that the Acmebot:SubscriptionId could be somehow selectable variable which would then apply when function is using Azure REST API to work with different App Services and Azure DNS zones?

This is currently kind of limitation, if we have several DNS zones, and multiple App services in multiple subscriptions using different DNS names. Acmebot is now the best in smaller environments, but many environments might also include multiple subscriptions and DNS zones.
